frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -734,14 +734,19 @@ func (c *cvdCommand) Run() error {
 	stderr := &bytes.Buffer{}
 	cmd.Stdout = c.stdout
 	cmd.Stderr = stderr
-	err := cmd.Run()
-	logStderr(cmd, stderr.String())
-	if ctx.Err() == context.DeadlineExceeded {
-		// The command is killed by a signal in these cases, but the real failure is the timeout.
-		return &cvdCommandTimeoutErr{c.args}
+	if err := cmd.Start(); err != nil {
+		return err
 	}
-	if err != nil {
-		return &cvdCommandExecErr{c.args, stderr.String(), err}
+	doneCh := make(chan error)
+	go func() { doneCh <- cmd.Wait() }()
+	select {
+	case err := <-doneCh:
+		logStderr(cmd, stderr.String())
+		if err != nil {
+			return &cvdCommandExecErr{c.args, stderr.String(), err}
+		}
+	case <-ctx.Done():
+		return &cvdCommandTimeoutErr{c.args}
```

Comment:
This doesn't stop the underlying process on timeout, potentially leaving background processes running indefinitely. Consider using [CommandContext](https://pkg.go.dev/os/exec#CommandContext) instead.

Reply:
yes, the caller would be the responsible to kill those processes. this is in coming commits. This commit solves the blocking issue only so the client  could get a "time out" response, which right now it does not get.

We are already using CommandContext, introduced here https://github.com/google/android-cuttlefish/commit/e3f20ad5d210ab7902c78c01d80e6e576f44c43e.

Reply:
1. The caller shouldn't be responsible of killing processes started by this function.
2. You can return "time out" by checking `context.Err()` after `Wait` returns an error, if it's `context.DeadlineExceeded` then return "timeout", otherwise it failed for some other reason. It can return `context.Cancelled` but there are no code paths here that would cause that.

Reply:
1. It's the caller that sets the cvd subcommand to be used, for example the caller using `cvd start` is aware that running `cvd start` might hang out leaving processes running after timeout therefore it should clean it up if time out happened. the caller using `cvd fleet` is aware no process will be left running.
2. Wait() or Run() blocks as stderr is passed to the `cvd_server` daemon is not closed. This is the reason we changed to  `select` to listen on `ctx.Done()` channel to avoid getting blocked on either `Run()` or `Wait()`. Here's more context on why it blocks https://github.com/google/android-cuttlefish/blob/main/frontend/src/host_orchestrator/orchestrator/instancemanager.go#L751-L756 even using a CommandContext with timeout.

Reply:
`Wait` won't block if you add a context with a timeout. When the context is done it will kill the process and `Wait` will return indicating it received signal 6 (abort) on linux. At least that what happened when I tested it with a time of 1 second and the command `sleep 20`.